pydra/engine/core.py

```diff
@@ -508,7 +508,7 @@ def add(self, task):
         self._last_added = task
         self.create_connections(task)
         logger.debug(f"Added {task}")
-        self.inputs._graph = deepcopy(self.graph_sorted)
+        self.inputs._graph_checksums = [nd.checksum for nd in self.graph_sorted]
```

Comment:
this would compute graph and node checksum multiple times for each add.

Reply:
the graph has to be recomputed (it's sorted)

Reply:
all i am saying is compute the checksum only when it is needed.